GitHub has test failure after merge

Seems random, only Ubuntu has failure.

https://github.com/moinwiki/moin/actions/runs/5501625909/jobs/10025342573

    ...
    -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html   
    =========================== short test summary info ============================
    FAILED src/moin/cli/_tests/test_modify_item.py::test_validate_metadata_missing_rev_num - KeyError: 'rev_number'
    = 1 failed, 1725 passed, 20 skipped, 2 xfailed, 1440 warnings in 342.44s (0:05:42) =
    Error: Process completed with exit code 1.

Swapped python versions per above, now have different failure on ubuntu prior to merge;

    =========================== short test summary info ============================
    FAILED src/moin/cli/_tests/test_modify_item.py::test_item_rev - AssertionError: assert 'MyPage version 1\r\n' == 'MyPage 
    version 2\r\n'
      - MyPage version 2
      ?                ^
      + MyPage version 1
      ?                ^
    = 1 failed, 1725 passed, 20 skipped, 2 xfailed, 1440 warnings in 375.17s (0:06:15) =

https://github.com/moinwiki/moin/actions/runs/5547582042/jobs/10129423062?pr=1475

Will merge to see what happens next.

Ubuntu only failure again:  https://github.com/moinwiki/moin/actions/runs/5579775538/jobs/10195787767?pr=1477

    =========================== short test summary info ============================
    FAILED src/moin/cli/_tests/test_modify_item.py::test_item_rev - AssertionError: assert 'MyPage version 1\r\n' == 'MyPage version 2\r\n'
      - MyPage version 2
      ?                ^
      + MyPage version 1
      ?                ^
    FAILED src/moin/cli/_tests/test_modify_item.py::test_validate_metadata_missing_rev_num - KeyError: 'rev_number'
    = 2 failed, 1724 passed, 20 skipped, 2 xfailed, 1440 warnings in 348.86s (0:05:48) =
    Error: Process completed with exit code 1.

Sometimes it breaks on windows:  https://github.com/RogerHaase/moin/actions/runs/5603630779/jobs/10250518035

    =========================== short test summary info ===========================
    FAILED src/moin/cli/_tests/test_modify_item.py::test_item_rev - AssertionError: assert 'MyPage version 1\r\n' == 'MyPage version 2\r\n'
      - MyPage version 2

      ?                ^
      + MyPage version 1

      ?                ^
    = 1 failed, 1725 passed, 20 skipped, 2 xfailed, 1440 warnings in 429.79s (0:07:09) =
    Error: Process completed with exit code 1.

I am always getting these failures now on my debian system: 

```
src/moin/cli/_tests/test_modify_item.py ..F...FF  
FAILED src/moin/cli/_tests/test_modify_item.py::test_dump_help - AssertionError: assert {'archive.tar...vg.data', ...} == {'archive.tar...ng.data', ...}
FAILED src/moin/cli/_tests/test_modify_item.py::test_validate_metadata - AssertionError: assert 'parentid_err...ion_number: 2' == 'parentid_err...ion_number: 1'
FAILED src/moin/cli/_tests/test_modify_item.py::test_validate_metadata_missing_rev_num - KeyError: 'rev_number'
```
My recommendation is to deactivate the failing tests for now. It will need some time to find the real cause.

--- a/src/moin/cli/_tests/test_modify_item.py
+++ b/src/moin/cli/_tests/test_modify_item.py
@@ -5,9 +5,9 @@
 MoinMoin - moin.cli.maint.modify_item tests
 """
 
-import pytest
 import json
 from pathlib import Path
+from time import sleep
 
 from moin._tests import get_dirs
 from moin.cli._tests import run, assert_p_succcess, read_index_dump_latest_revs, read_index_dump
@@ -36,7 +36,6 @@ def test_welcome(welcome):
     assert_p_succcess(welcome)
 
 
-@pytest.mark.skip(reason="1471 fails on debian")
 def test_dump_help(load_help):
     moin_dir, artifact_dir = get_dirs('cli')
     help_dir = Path('my_help')
@@ -111,7 +110,6 @@ def test_item_put(index_create2):
     assert ["русский"] == my_item['tags']
 
 
-@pytest.mark.skip(reason="1471 fails after github merge")
 def test_item_rev(index_create2):
     """test loading multiple versions of same page
 
@@ -126,6 +124,7 @@ def test_item_rev(index_create2):
     data_dir = moin_dir / 'src' / 'moin' / 'cli' / '_tests' / 'data'
     put1 = run(['moin', 'item-put', '-m', data_dir / 'MyPage-v1.meta', '-d', data_dir / 'MyPage-v1.data', '-o'])
     assert_p_succcess(put1)
+    sleep(1.1)  # MTIME is stored as int in index file, sleep here to guarntee proper sorting of revs for index_revision
     put2 = run(['moin', 'item-put', '-m', data_dir / 'MyPage-v2.meta', '-d', data_dir / 'MyPage-v2.data', '-o'])
     assert_p_succcess(put2)
     item_get2 = run(['moin', 'item-get', '-n', 'MyPage', '-m', 'MyPage-v2.meta', '-d', 'MyPage-v2.data', '--crlf'])
@@ -155,12 +154,12 @@ def test_item_rev(index_create2):
     assert v1_1_meta['size'] == 16  # validate no newline at end in storage
 
 
-@pytest.mark.skip(reason="1471 fails on debian")
 def test_validate_metadata(index_create2):
     moin_dir, _ = get_dirs('')
     data_dir = moin_dir / 'src' / 'moin' / 'cli' / '_tests' / 'data'
     item_put = run(['moin', 'item-put', '-m', data_dir / 'MyPage-v1.meta', '-d', data_dir / 'MyPage-v1.data', '-o'])
     assert_p_succcess(item_put)
+    sleep(1.1)  # MTIME is stored as int in index file, sleep here to guarntee proper sorting of revs for index_revision
     item_put = run(['moin', 'item-put', '-m', data_dir / 'MyPage-v2.meta', '-d', data_dir / 'MyPage-v2.data', '-o'])
     assert_p_succcess(item_put)
     validate = run(['moin', 'maint-validate-metadata', '-b', 'default', '-v'])
@@ -170,8 +169,10 @@ def test_validate_metadata(index_create2):
     assert b'0 items with invalid metadata found' == outlines[0]
     item_put = run(['moin', 'item-put', '-m', data_dir / 'Corrupt.meta', '-d', data_dir / 'Corrupt.data', '-o'])
     assert_p_succcess(item_put)
+    sleep(1.1)  # MTIME is stored as int in index file, sleep here to guarntee proper sorting of revs for index_revision
     item_put = run(['moin', 'item-put', '-m', data_dir / 'Corrupt2.meta', '-d', data_dir / 'Corrupt2.data', '-o'])
     assert_p_succcess(item_put)
+    sleep(1.1)  # MTIME is stored as int in index file, sleep here to guarntee proper sorting of revs for index_revision
     item_put = run(['moin', 'item-put', '-m', data_dir / 'Corrupt3.meta', '-d', data_dir / 'Corrupt3.data', '-o'])
     assert_p_succcess(item_put)
     validate = run(['moin', 'maint-validate-metadata', '-b', 'default', '-v'])
@@ -217,6 +218,7 @@ def test_validate_metadata(index_create2):
     assert 11 == metas[rev_id1][SIZE]
     assert PARENTID not in metas[rev_id3]
     # create a repeated revision_number
+    sleep(1.1)  # MTIME is stored as int in index file, sleep here to guarntee proper sorting of revs for index_revision
     item_put = run(['moin', 'item-put', '-m', data_dir / 'MyPage-v2.meta', '-d', data_dir / 'MyPage-v2.data'])
     assert_p_succcess(item_put)
     validate = run(['moin', 'maint-validate-metadata', '-b', 'default', '-v', '-f'])
@@ -236,12 +238,12 @@ def test_validate_metadata(index_create2):
     assert rev_numbers[2][REVID] == rev_id5
 
 
-@pytest.mark.skip(reason="1471 fails on debian and after github merge on ubuntu")
 def test_validate_metadata_missing_rev_num(index_create2):
     moin_dir, _ = get_dirs('')
     data_dir = moin_dir / 'src' / 'moin' / 'cli' / '_tests' / 'data'
     item_put = run(['moin', 'item-put', '-m', data_dir / 'MyPage-vblank.meta', '-d', data_dir / 'MyPage-v1.data', '-o'])
     assert_p_succcess(item_put)
+    sleep(1.1)  # MTIME is stored as int in index file, sleep here to guarntee proper sorting of revs for index_revision
     item_put = run(['moin', 'item-put', '-m', data_dir / 'MyPage-vblank2.meta', '-d', data_dir / 'MyPage-v1.data', '-o'])
     assert_p_succcess(item_put)
     validate = run(['moin', 'maint-validate-metadata', '-b', 'default', '-v', '-f'])

--- a/src/moin/cli/_tests/test_scrapy_crawl.py
+++ b/src/moin/cli/_tests/test_scrapy_crawl.py
@@ -61,7 +61,6 @@ class TestSiteCrawl:
         expected = CrawlResultMatch(url='/Home')
         assert expected.match(r), f'unexpected redirect for / {r}'
 
-    @pytest.mark.skip
     def test_200(self, crawl_results):
         assert crawl_results[1], f'crawl failed, check {get_crawl_log_path()}'
         failures = []

